packages/menu-bar/src/vaadin-menu-bar-mixin.js

```diff
@@ -759,12 +759,7 @@ export const MenuBarMixin = (superClass) =>
      */
     _setFocused(focused) {
       if (focused) {
-        let target = this.querySelector('[tabindex="0"]');
-        if (this.tabNavigation) {
-          // Switch submenu on menu button Tab / Shift Tab
-          target = this.querySelector('[focused]');
-          this.__switchSubMenu(target);
-        }
```

Comment:
This logic was previously switching submenu for root level buttons <kbd>Tab</kbd> / <kbd>Shift</kbd> + <kbd>Tab</kbd>.